File: 08.Week-08/35.Day-35_HandsOn_Contact_Mgmt_Using_Dapper/Services/ProductService.cs
using WebApplication5.Models;
using WebApplication5.Repository;

namespace WebApplication5.Services
{
    public class ProductService : IProductService
    {
        private readonly IProductRepository _repository;

        public ProductService(IProductRepository repository)
        {
            _repository = repository;
        }
        public IEnumerable<Product> GetProducts()
        {
            return _repository.GetAll();
        }
        public Product GetProduct(int id)
        {
            return _repository.GetById(id);
        }
        public void CreateProduct(Product product)
        {
            _repository.Add(product);
        }

        public void UpdateProduct(Product product)
        {
            _repository.Update(product);
        }

        public void DeleteProduct(int id)
        {
            _repository.Delete(id);
        }
    }
    }

File: 12.Week-12/Day58_HandsOn/Angulardocker/src/app/services/product-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Product } from '../models/Product';
 

@Injectable({
  providedIn: 'root'  // Project level and singleton 
})
export class ProductService {

    readonly API_URL:string = "http://localhost:5000/api/Product";

    constructor(private httpClient:HttpClient){}

    getProducts() {
      return this.httpClient.get<Product[]>(this.API_URL);
    }

    getProductById(id:number) {
      return this.httpClient.get<Product>(this.API_URL+ id);
    }

    addProduct(product:Product)     {
      return this.httpClient.post(this.API_URL, product);
    }

  updateProduct(id: number, product: Product) {
  return this.httpClient.put(this.API_URL + id, product);
}

deleteProduct(id: number) {
  return this.httpClient.delete(this.API_URL+ id); 
}

}